src/wallet/init.cpp

```diff
@@ -183,22 +171,6 @@ bool WalletInit::ParameterInteraction()
                         _("This is the transaction fee you may discard if change is smaller than dust at this level"));
         CWallet::m_discard_rate = CFeeRate(nFeePerK);
     }
-    if (gArgs.IsArgSet("-paytxfee"))
```

Comment:
Is there a reason not to make options like -discardfee and -mintxfee wallet members, the same way this is doing for -fallbackfee and -paytxfee options? It doesn't seem like a good thing to create a distinction here when moving the options together is possible.

Making more options members also seems nice in case we want to add a wallet RPC for   changing preferences dynamically without affecting other wallets.

Reply:
mintxfee et al are already members (static, though). I agree that those should be changed to be non-static members, but I'd rather do this in a follow up pr. Otherwise this one will explode in scope, scare away revierers and will gracelessly end up in rebase hell.

Reply:
> Making more options members also seems nice in case we want to add a wallet RPC for changing preferences dynamically without affecting other wallets.

Please see the included test case: We are already doing it wrong.

Reply:
https://github.com/bitcoin/bitcoin/pull/12909#discussion_r180244428

> mintxfee et al are already members (static, though). I agree that those should be changed to be non-static members, but I'd rather do this in a follow up pr. Otherwise this one will explode in scope, scare away revierers and will gracelessly end up in rebase hell.

I think "wallet-specific fee options" would be an appropriate scope for this PR and not too broad. The -mintxfee, -fallbackfee, -discardfee, and -paytxfee options are literally initialized one after another here, so it's strange to me that you want to move the second and fourth initializations but not the first and third based them on being global variables instead of static class member variables (since all the variables function identically).